plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetescluster/KubernetesClusterManagerImpl.java

```diff
@@ -0,0 +1,3061 @@
+        kubernetesClusterDao.update(kubernetesCluster.getId(), kubernetesCluster);
+
+        int sshPort = publicIpSshPort.second();
+        boolean readyNodesCountValid = validateKubernetesClusterReadyNodesCount(kubernetesCluster, publicIpAddress, sshPort, 30, 30000);
+
+        // Detach binaries ISO from new VMs
+        detachIsoKubernetesVMs(kubernetesCluster, clusterVMIds);
+
+        // Throw exception if nodes count for k8s cluster timed out
+        if (!readyNodesCountValid) { // Scaling failed
+            String msg = String.format("Failed to setup Kubernetes cluster ID: %s as it does not have desired number of nodes in ready state", kubernetesCluster.getUuid());
+            LOGGER.warn(msg);
+            stateTransitTo(kubernetesCluster.getId(), KubernetesCluster.Event.CreateFailed);
+            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, msg);
+        }
+
+        boolean k8sKubeConfigCopied = false;
+        String kubeConfig = getKubernetesClusterConfig(kubernetesCluster, publicIpAddress, sshPort, 5);
+        if (!Strings.isNullOrEmpty(kubeConfig)) {
+            k8sKubeConfigCopied = true;
+        }
+        if (!k8sKubeConfigCopied) {
+            String msg = String.format("Failed to setup Kubernetes cluster ID: %s in usable state as unable to retrieve kube-config for the cluster", kubernetesCluster.getUuid());
+            LOGGER.error(msg);
+            stateTransitTo(kubernetesCluster.getId(), KubernetesCluster.Event.OperationFailed);
+            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, msg);
+        }
+        kubeConfig = kubeConfig.replace(String.format("server: https://%s:%d", k8sMasterVM.getPrivateIpAddress(), CLUSTER_API_PORT),
+                String.format("server: https://%s:%d", publicIpAddress, CLUSTER_API_PORT));
+        kubernetesClusterDetailsDao.addDetail(kubernetesCluster.getId(), "kubeConfigData", Base64.encodeBase64String(kubeConfig.getBytes(Charset.forName("UTF-8"))), false);
+
+        boolean dashboardServiceRunning = isKubernetesClusterDashboardServiceRunning(kubernetesCluster, publicIpAddress, sshPort, 10, 20000);
+        if (!dashboardServiceRunning) {
+            String msg = String.format("Failed to setup Kubernetes cluster ID: %s in usable state as unable to get Dashboard service running for the cluster", kubernetesCluster.getUuid());
+            LOGGER.error(msg);
+            stateTransitTo(kubernetesCluster.getId(), KubernetesCluster.Event.OperationFailed);
+            throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, msg);
+        }
+        kubernetesClusterDetailsDao.addDetail(kubernetesCluster.getId(), "dashboardServiceRunning", String.valueOf(dashboardServiceRunning), false);
+        stateTransitTo(kubernetesCluster.getId(), KubernetesCluster.Event.OperationSucceeded);
+        return true;
+    }
+
+    private boolean startStoppedKubernetesCluster(long kubernetesClusterId) throws ManagementServerException,
```

Comment:
also a hundred lines. can you please factor out some of that code for readability?